rasa_nlu/server.py

```diff
@@ -277,29 +272,73 @@ def status(self, request):
         request.setHeader('Content-Type', 'application/json')
         return json_to_string(self.data_router.get_status())
 
+    def extract_json(self, content):
+        # test if json has config structure
+        json_config = simplejson.loads(content).get("data")
+
+        # if it does then this results in correct format.
+        if json_config:
+
+            model_config = simplejson.loads(content)
+            data = json_config
+
+        # otherwise use defaults.
+        else:
+
+            model_config = self.default_model_config
+            data = content
+
+        return model_config, data
+
+    def extract_data_and_config(self, request):
+
+        request_content = request.content.read().decode('utf-8', 'strict')
+        content_type = self.get_request_content_type(request)
+
+        if 'yml' in content_type:
+            # assumes the user submitted a model configuration with a data
+            # parameter attached to it
+
+            model_config = utils.read_yaml(request_content)
+            data = model_config.get("data")
+
+        elif 'json' in content_type:
+            # If the caller just provided training data without config
+            # this will use the default model config the server
+            # was started with
+
+            model_config, data = self.extract_json(request_content)
+
+        else:
+
+            raise Exception("Content-Type must be yml or json")
+
+        return model_config, data
+
+    def get_request_content_type(self, request):
+        content_type = request.requestHeaders.getRawHeaders("Content-Type", [])
```

Comment:
Does this still work fine when content type isn't set?

Reply:
I handled it. Raise exception, must have exactly 1.